examples_tests/10.AllocatorTest/main.cpp

```diff
@@ -0,0 +1,420 @@
+		{
+			switch (event.KeyInput.Key)
+			{
+			case irr::KEY_KEY_Q: // switch wire frame mode
+				exit(0);
+				return true;
+			default:
+				break;
+			}
+		}
+
+		return false;
+	}
+
+private:
+};
+
+constexpr size_t minTestsCnt = 10000u;
+constexpr size_t maxTestsCnt = 20000u;
+constexpr size_t maxAlignmentExp = 12u;                         // 4096
+constexpr size_t minVirtualMemoryBufferSize = 2048;             // 2kB
+constexpr size_t maxVirtualMemoryBufferSize = 2147483648;       // 2GB
+
+class RandomNumberGenerator
+{
+public:
+	RandomNumberGenerator()
+		: mt(rd())
+	{
+	}
+
+	inline uint32_t getRndAllocCnt()    { return  allocsPerFrameRange(mt);  }
+	inline uint32_t getRndMaxAlign()    { return  (1u << maxAlignmentExpPerFrameRange(mt)); } //4096 is max
+	inline uint32_t getRndBuffSize()    { return  buffSzRange(mt); }
+
+	inline uint32_t getRandomNumber(uint32_t rangeBegin, uint32_t rangeEnd)   
+	{
+		std::uniform_int_distribution<uint32_t> dist(rangeBegin, rangeEnd);
+		return dist(mt);
+	}
+
+	inline std::mt19937& getMt()
+	{
+		return mt;
+	}
+
+private:
+	std::random_device rd;
+	std::mt19937 mt;
+	const std::uniform_int_distribution<uint32_t> allocsPerFrameRange = std::uniform_int_distribution<uint32_t>(minTestsCnt, maxTestsCnt);
+	const std::uniform_int_distribution<uint32_t> maxAlignmentExpPerFrameRange = std::uniform_int_distribution<uint32_t>(1, maxAlignmentExp);
+	const std::uniform_int_distribution<uint32_t> buffSzRange = std::uniform_int_distribution<uint32_t>(minVirtualMemoryBufferSize, maxVirtualMemoryBufferSize);
+};
+
+RandomNumberGenerator rng;
+
+template<typename AlctrType>
+class AllocatorHandler
+{
+	using Traits = core::address_allocator_traits<AlctrType>;
+
+public:
+	void executeAllocatorTest()
+	{
+		const uint32_t testsCnt = rng.getRndAllocCnt();
+		for (size_t i = 0; i < testsCnt; ++i)
+			executeForFrame();
+	}
+
+private:
+	struct AllocationData
+	{
+		uint32_t outAddr = AlctrType::invalid_address;
+		uint32_t size = 0u;
+		uint32_t align = 0u;
+	};
+
+	struct RandParams
+	{
+		uint32_t maxAlign;
+		uint32_t addressSpaceSize;
+		uint32_t alignOffset;
+		uint32_t offset;
+		uint32_t blockSz;
+	};
+
+private:
+	void executeForFrame()
+	{
+		RandParams randAllocParams = getRandParams();
+		void* reservedSpace = nullptr;
+		AlctrType alctr;
+
+		if constexpr (std::is_same<AlctrType, core::LinearAddressAllocator<uint32_t>>::value)
+		{
+			alctr = AlctrType(nullptr, randAllocParams.offset, randAllocParams.alignOffset, randAllocParams.maxAlign, randAllocParams.addressSpaceSize);
+		}
+		else
+		{
+			const auto reservedSize = AlctrType::reserved_size(randAllocParams.maxAlign, randAllocParams.addressSpaceSize, randAllocParams.blockSz);
+			reservedSpace = _NBL_ALIGNED_MALLOC(reservedSize, _NBL_SIMD_ALIGNMENT);
+			alctr = AlctrType(reservedSpace, randAllocParams.offset, randAllocParams.alignOffset, randAllocParams.maxAlign, randAllocParams.addressSpaceSize, randAllocParams.blockSz);
+		}
```

Comment:
this should move to `executeAllocatorTest` 

because right now you're only testing `reset` at most once per allocator creation (you dont test reuse of allocator after reset)